commit 5b9bd473e3b8a8c6c4ae99be475e6e9b27568555
Author: Oleg Nesterov <oleg@redhat.com>
Date:   Fri Mar 23 15:02:49 2012 -0700

    usermodehelper: ____call_usermodehelper() doesn't need do_exit()
    
    Minor cleanup.  ____call_usermodehelper() can simply return, no need to
    call do_exit() explicitely.
    
    Signed-off-by: Oleg Nesterov <oleg@redhat.com>
    Cc: Tetsuo Handa <penguin-kernel@I-love.SAKURA.ne.jp>
    Cc: Rusty Russell <rusty@rustcorp.com.au>
    Cc: Tejun Heo <tj@kernel.org>
    Cc: David Rientjes <rientjes@google.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
    [dannf: adjusted to apply to Debian's 2.6.32]

diff --git a/kernel/kmod.c b/kernel/kmod.c
index 0c775dc..8ed592b 100644
--- a/kernel/kmod.c
+++ b/kernel/kmod.c
@@ -190,7 +190,7 @@ static int ____call_usermodehelper(void *data)
 
 	/* Exec failed? */
 	sub_info->retval = retval;
-	do_exit(0);
+	return 0;
 }
 
 void call_usermodehelper_freeinfo(struct subprocess_info *info)
